### PR TITLE
[Microsoft.ApiManagement][2020-06-01-preview] Gateway certificate authorities

### PR DESCRIPTION
**Pull request by maksimkim**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/examples/ApiManagementListGatewayCertificateAuthorities.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          {
+            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/gateways/gw1/certificateAuthorities/cert2",
+            "type": "Microsoft.ApiManagement/service/gateways/certificateAuthorities",
+            "name": "cert2",
+            "properties": {
+              "isTrusted": true
+            }
+          }
+        ],
+        "nextLink": ""
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/examples/ApiManagementListGatewayCertificateAuthorities.json:30]: Do you have a `count` parameter here?

**Review by solankisamir — APPROVED** [review]
One nit comment on the `count`

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimgateways.json [hunk 2]
```diff
@@ -1029,6 +1029,295 @@
...
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/GatewayIdParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "|     Field     |     Usage     |     Supported operators     |     Supported functions     |</br>|-------------|-------------|-------------|-------------|</br>| name | filter | eq, ne |  |</br>"
```
Comment (yungezz) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimgateways.json:1060]: not sure will this cause problem when building docs at docs.ms

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json [hunk 3]
```diff
@@ -5866,6 +5866,48 @@
...
+          "description": "Next page link if any.",
+          "readOnly": true
+        }
+      },
+      "description": "Paged Gateway certificate authority list representation."
+    },
+    "GatewayCertificateAuthorityContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
```
Comment (yungezz) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:5890]: just curious, why need this?

**Comment by yungezz** [comment]
waiting for ARM feedback since new APIs added.

**Comment by yungezz** [comment]
hi @msyyc, could you pls confirm impact to Python SDK? thanks

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimgateways.json [hunk 4]
```diff
@@ -1029,6 +1029,295 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
```
Comment (jorgecotillo) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimgateways.json:1197]: >"put" [](start = 6, length = 5)

Please add x-ms-long-running-operation as stated [here](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation)

Comment (jorgecotillo) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimgateways.json:1197]: >"put" [](start = 6, length = 5)

just making sure, is this a sync operation right?

Comment (solankisamir) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimgateways.json:1197]: yes

**Comment by solankisamir** [comment]
@yungezz can you take a look again? This is signed off from ARM and Breaking Change team

**Comment by yungezz** [comment]
> hi @msyyc, could you pls confirm impact to Python SDK? thanks
hi @msyyc, could you pls confirm impact to Python SDK? thanks

**Comment by solankisamir** [comment]
@yungezz we have not released python sdk for this preview api-version

**Comment by yungezz** [comment]
> @yungezz we have not released python sdk for this preview api-version

thanks. could you pls fix CI failure? and is the change deployed to Prod already?

**Comment by solankisamir** [comment]
Copy of the changes was merged

**Pull request closed without merging**